scalameter-core/src/main/scala/org/scalameter/InvocationCounting.scala

```diff
@@ -0,0 +1,64 @@
+package org.scalameter
+
+import java.io.File
+import scala.collection.Seq
+import org.scalameter.execution.invocation.InvocationCountMatcher
+import org.scalameter.execution.invocation.instrumentation.{Instrumentation, MethodInvocationCounter}
+import org.scalameter.Key._
+
+
+/** Mixin for all [[org.scalameter.Measurer]] implementations that perform any kind of method invocation counting. */
+trait InvocationCounting extends Measurer {
+  def matcher: InvocationCountMatcher
+
+  def units: String = "#"
+
+  def measure[T, U](context: Context, measurements: Int, setup: (T) => Any, tear: (T) => Any, regen: () => T, snippet: (T) => Any): Seq[Double] = {
+    var iteration = 0
+    val invocationsBuilder = List.newBuilder[Double]
+    var value: T = null.asInstanceOf[T]
+    var obj: Any = null.asInstanceOf[Any]
+    val numMethods = context.goe(exec.measurers.methodInvocationLookupTable,
+      sys.error("Measurer.prepareContext must be called before Measurer.measure")).length
+
+    while (iteration < measurements) {
+      value = regen()
+
+      MethodInvocationCounter.setup(numMethods)
+      setup(value)
+
+      MethodInvocationCounter.start()
+      obj = snippet(value)
+      MethodInvocationCounter.stop()
+      tear(value)
+
+      invocationsBuilder += MethodInvocationCounter.counts.sum
+      iteration += 1
+    }
+
+    val invocations = invocationsBuilder.result()
+    log.verbose("Measurements: " + invocations.mkString(", "))
+    invocations
+  }
+
+  /** Creates a jar with instrumented classes, prepends it to a classpath key and saves the internal method lookup table. */
+  override def prepareContext(context: Context): Context = {
+    val jar = File.createTempFile(s"scalameter-bench-", "-instrumented.jar")
+    jar.deleteOnExit()
+
+    val lookupTable = Instrumentation.writeInstrumentedClasses(ctx = context, matcher = matcher, to = jar)
+    val classPath = context.goe(classpath, utils.ClassPath.default)
+    Context(exec.measurers.methodInvocationLookupTable -> lookupTable, classpath -> (jar +: classPath))
```

Comment:
I don't know about any other methods for setting `ClassLoder` in runtime then `setContextClassLoader` from the `Thread` and I don't know if this would work in that case. 

So to be sure that it won't block other parts maybe I could check it later, when we iterate over this PR?